datasets/ru/cbr_banks/crawler.py

```diff
@@ -0,0 +1,189 @@
+    "Cache-Control": "max-age=0",
+    "Connection": "keep-alive",
+    "Cookie": "__ddg2_=oODOe2gVFAHstFSz; __ddg5_=5VLoRhvMkPVMHI4H; __ddgid_=F69Zw7VH7FdLn3xt; __ddgmark_=C04rsUQPh6Z7r9Xw; _ym_d=1730798853; _ym_isad=2; _ym_uid=1730798853821838136; __ddg1_=FwX17Njle4Mnl9WOR8V5",
+    "Host": "cbr.ru",
+    "Referer": "https://cbr.ru/scripts/XML_bic2.asp",
+    "Sec-Fetch-Dest": "document",
+    "Sec-Fetch-Mode": "navigate",
+    "Sec-Fetch-Site": "same-origin",
+    "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.6 Safari/605.1.15",
+}
+
+SOAP_URL = "http://www.cbr.ru/CreditInfoWebServ/CreditOrgInfo.asmx"
+SOAP_HEADERS = {"Content-Type": "text/xml; charset=utf-8"}
+NAMESPACE = {
+    "soap": "http://schemas.xmlsoap.org/soap/envelope/",
+    "ns": "http://web.cbr.ru/",
+}
+
+
+def send_soap_request(action, body):
+    """Sends a SOAP request and returns the parsed XML response."""
+    headers = SOAP_HEADERS.copy()
+    headers["SOAPAction"] = f"http://web.cbr.ru/{action}"
+
+    response = requests.post(SOAP_URL, data=body, headers=headers)
+    if response.status_code != 200:
+        raise Exception(
+            f"Request failed with status {response.status_code}: {response.text}"
+        )
+
+    return etree.fromstring(response.content)
+
+
+def bic_to_int_code(bic):
+    """Gets the internal code for a BIC."""
+    # Formulate the SOAP request body
+    body = f"""<?xml version="1.0" encoding="utf-8"?>
+    <soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+        <soap:Body>
+            <BicToIntCode xmlns="http://web.cbr.ru/">
+                <BicCode>{bic}</BicCode>
+            </BicToIntCode>
+        </soap:Body>
+    </soap:Envelope>"""
+
+    tree = send_soap_request("BicToIntCode", body)
+
+    # Extract the result
+    result = tree.find(".//ns:BicToIntCodeResult", namespaces=NAMESPACE)
+    if result is not None:
+        return result.text
+    else:
+        print("Result not found in the response")
+        return None
+
+
+def details_by_int_code(internal_code):
+    """Gets detailed credit information for an internal code."""
+    # Create the SOAP request body
+    body = f"""<?xml version="1.0" encoding="utf-8"?>
+    <soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+        <soap:Body>
+            <CreditInfoByIntCodeXML xmlns="http://web.cbr.ru/">
+                <InternalCode>{internal_code}</InternalCode>
+            </CreditInfoByIntCodeXML>
+        </soap:Body>
+    </soap:Envelope>"""
+
+    tree = send_soap_request("CreditInfoByIntCodeXML", body)
+
+    details = {}
+    # Locate the main result element
+    result = tree.find(".//ns:CreditInfoByIntCodeXMLResult", namespaces=NAMESPACE)
```

Comment:
you can use return h.remove_namespace(root) to remove the tag namespaces and make it a lot easier to query